clients/src/main/java/org/apache/kafka/clients/admin/DescribeUserScramCredentialsResult.java

```diff
@@ -0,0 +1,145 @@
+     * include users that have a credential but that could not be described.
+     */
+    public KafkaFuture<List<String>> users() {
+        return KafkaFuture.allOf(dataFuture).thenApply(v -> {
+            DescribeUserScramCredentialsResponseData data = valueFromFutureGuaranteedToSucceedAtThisPoint(dataFuture);
+            return data.results().stream()
+                    .filter(result -> result.errorCode() != Errors.RESOURCE_NOT_FOUND.code())
+                    .map(result -> result.user()).collect(Collectors.toList());
+        });
+    }
+
+    /**
+     *
+     * @param userName the name of the user description being requested
+     * @return a future indicating the description results for the given user. The future will complete exceptionally if
+     * the future returned by {@link #users()} completes exceptionally.  Note that if the given user does not exist in
+     * the list of described users then the returned future will complete exceptionally with
+     * {@link org.apache.kafka.common.errors.ResourceNotFoundException}.
+     */
+    public KafkaFuture<UserScramCredentialsDescription> description(String userName) {
+        return KafkaFuture.allOf(dataFuture).thenApply(v -> {
+            DescribeUserScramCredentialsResponseData data = valueFromFutureGuaranteedToSucceedAtThisPoint(dataFuture);
+            // it is possible that there is no future for this user (for example, the original describe request was for
+            // users 1, 2, and 3 but this is looking for user 4), so explicitly take care of that case
+            Optional<DescribeUserScramCredentialsResponseData.DescribeUserScramCredentialsResult> optionalUserResult =
+                    data.results().stream().filter(result -> result.user().equals(userName)).findFirst();
+            if (!optionalUserResult.isPresent()) {
+                throw new ResourceNotFoundException("No such user: " + userName);
+            }
+            DescribeUserScramCredentialsResponseData.DescribeUserScramCredentialsResult userResult = optionalUserResult.get();
+            if (userResult.errorCode() != Errors.NONE.code()) {
+                // RESOURCE_NOT_FOUND is included here
+                throw Errors.forCode(userResult.errorCode()).exception(userResult.errorMessage());
+            }
+            return new UserScramCredentialsDescription(userResult.user(), getScramCredentialInfosFor(userResult));
+        });
+    }
+
+    private static List<ScramCredentialInfo> getScramCredentialInfosFor(
+            DescribeUserScramCredentialsResponseData.DescribeUserScramCredentialsResult userResult) {
+        return userResult.credentialInfos().stream().map(c ->
+                new ScramCredentialInfo(ScramMechanism.fromType(c.mechanism()), c.iterations()))
+                .collect(Collectors.toList());
+    }
+
+    private static <T> T valueFromFutureGuaranteedToSucceedAtThisPoint(KafkaFuture<T> future) {
+        try {
+            return future.get();
```

Comment:
The call to get here isn't needed.  See the comment about about using `KafkaFuture#whenComplete`.  Chaining future results is a big part of the power of `CompletableFuture` (which `KafkaFuture` is a clone of... long story...)